auth/src/main/java/org/aerogear/mobile/auth/user/UserPrincipalImpl.java

```diff
@@ -37,10 +47,14 @@
      */
     protected UserPrincipalImpl(final String username,
                               final String email,
-                              final Set<UserRole> roles) {
+                              final Set<UserRole> roles,
+                              final String identityToken,
+                              final String accessToken) {
         this.username = username;
         this.email = email;
         this.roles = Collections.synchronizedSet(new HashSet<>(roles));
```

Comment:
Why synchronizedSet?

Reply:
I think I added this in cause I thought that a set needed to be synchronised when iterating over that set.  The set is being iterated over for the toString() function here -> https://github.com/aerogear/aerogear-android-sdk/blob/master/auth/src/main/java/org/aerogear/mobile/auth/user/UserPrincipalImpl.java#L131

Reply:
It is needed only if you need to handle concurrent modifications. Otherwise it does not need to be synchronized.
Better to remove that.

Reply:
Agreed. Will remove it.